src/nemos/basis.py

```diff
@@ -1202,15 +1206,15 @@ def _check_n_basis_min(self) -> None:
         Raises
         ------
         ValueError
-            If an insufficient number of basis element is requested for the basis type
+            If an insufficient number of basis element is requested for the basis type.
         """
         if self.n_basis_funcs < 1:
             raise ValueError(
-                f"Object class {self.__class__.__name__} requires >= 1 basis elements. "
+                f"Object class {self.__class__.__name__} requires >= 0 basis elements. "
```

Comment:
I see the error message now matches the check, but shouldn't this actually at least 1 basis element? With `max_freq=0` (just the DC term), you get a `n_basis_funcs=1`. And it doesn't make sense to have `max_freq<0`